src/core/solvers/ddp.cpp

```diff
@@ -246,8 +246,13 @@ void SolverDDP::forwardPass(const double& steplength) {
   for (std::size_t t = 0; t < T; ++t) {
     const boost::shared_ptr<ActionModelAbstract>& m = problem_->get_runningModels()[t];
     const boost::shared_ptr<ActionDataAbstract>& d = problem_->get_runningDatas()[t];
-
-    m->get_state()->diff(xs_[t], xs_try_[t], dx_[t]);
+    
+    if (t == 0) {
+      m->get_state()->diff(problem_->get_x0(), xs_[0], dx_[0]);
```

Comment:
I think it is better if you don't specialize for t==0 here. Rather, you should look at 
1) https://github.com/loco-3d/crocoddyl/blob/devel/src/core/solvers/ddp.cpp#L43
2) https://github.com/loco-3d/crocoddyl/blob/devel/src/core/solvers/ddp.cpp#L155
3) https://github.com/loco-3d/crocoddyl/blob/devel/src/core/solvers/ddp.cpp#L345
At all these places, we are assigining xs_try_ to be equal to get_x0(). Instead, we need to make sure that dx0 is properly accounted for in these assignments.